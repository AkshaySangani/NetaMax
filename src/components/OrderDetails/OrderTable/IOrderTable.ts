import { IOrderDetails } from "dataflows/OrderDetails/IOrderDetails";

export type IOrderTableProps = IOrderDetails;
