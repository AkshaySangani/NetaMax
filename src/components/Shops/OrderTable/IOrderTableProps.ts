import { IItem } from "dataflows/Shops/IItem";

export interface IOrderTableProps {
  /**
   * The Subtotal Order
   * @type {Number}
   *
   * */
  subtotal: number;

  /**
   * The Discount Order
   * @type {Number}
   *
   * */
  discount: number;

  /**
   * The total Order
   * @type {Number}
   *
   * */
  total: number;

  /**
   * The order's item array
   * @type {IItem}
   *
   * */
  objects: IItem[];
}
