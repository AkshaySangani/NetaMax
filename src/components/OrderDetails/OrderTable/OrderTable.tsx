import React, { ReactElement } from "react";

import { CURRENCY_NAME } from "constants/localeFormatConstants";
import { IOrderItem } from "dataflows/OrderDetails/IOrderItem";
import { formatMoney } from "utils/currencyUtils";

import { Image } from "@chakra-ui/image";
import { Box, Container, Divider, Flex, Grid, GridItem, Text } from "@chakra-ui/layout";
import { Table, Tbody, Td, Tr } from "@chakra-ui/react";

import { IOrderTableProps } from "./IOrderTable";

/**
 * The order table component.
 * @param {IOrderTableProps} props the order table props.
 * @returns { ReactElement } The table component
 *
 */
export const OrderTable = (props: IOrderTableProps): ReactElement => {
  const { items, order_subtotal_excl_tax, order_discount, order_total } = props;

  return (
    <div>
      <Container
        centerContent={false}
        maxW="976px"
        boxShadow="md"
        rounded="md"
        bg="white"
        display={["none", "block", "block", "block"]}
        mt="2"
      >
        <Box maxHeight="480px" overflow="auto">
          <Table variant="simple">
            <Tbody>
              {items.map((item: IOrderItem) => (
                <Tr key={item.product_id}>
                  <Td>
                    <Box display="flex">
                      {item.image && (
                        <Image borderRadius="lg" width="40px" src={item.image} mr="4" />
                      )}
                      <Text fontWeight="700">{item.name}</Text>
                    </Box>
                  </Td>
                  <Td>
                    <Text fontWeight="700">Cantidad </Text>
                    {item.quantity}
                  </Td>
                  <Td>
                    <Text fontWeight="700">Precio unitario </Text>
                    {formatMoney(item.unit_price_incl_tax, CURRENCY_NAME)}
                  </Td>
                </Tr>
              ))}
            </Tbody>
          </Table>
        </Box>
        <Box align="right" mr="8" mt="4" pb="4">
          <Text fontWeight="500" fontSize="20px" color="#3870FF">
            {"Subtotal: " + formatMoney(order_subtotal_excl_tax, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="20px" color="#F54A49">
            {"Descuento: " + formatMoney(order_discount, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="27px" color="#3870FF">
            {"Total: " + formatMoney(order_total, CURRENCY_NAME)}
          </Text>
        </Box>
      </Container>
      <Box display={["block", "none", "none", "none"]}>
        <Box maxHeight="500px" overflow="auto">
          {items.map((item: IOrderItem) => (
            <Box key={item.product_id} mb="4">
              <Divider mb="4" borderColor="#9F9F9F" />
              <Flex ml="4">
                <Box display="flex">
                  {item.image && <Image borderRadius="lg" width="40px" src={item.image} mr="4" />}
                  <Text fontWeight="700" fontSize="16px">
                    {item.name}
                  </Text>
                </Box>
              </Flex>
              <Grid ml="16" mt="2" h="40px" templateColumns="repeat(4, 1fr)">
                <GridItem colSpan={2}>
                  <Text fontWeight="700" fontSize="15px">
                    Cantidad
                  </Text>
                </GridItem>
                <GridItem colSpan={2}>
                  <Text fontWeight="700" fontSize="15px">
                    Precio
                  </Text>
                </GridItem>
                <GridItem colSpan={2}>
                  <Text fontSize="15px">{item.quantity}</Text>
                </GridItem>
                <GridItem colSpan={2}>
                  <Text fontSize="15px">
                    {formatMoney(item.unit_price_incl_tax, CURRENCY_NAME)}
                  </Text>
                </GridItem>
              </Grid>
            </Box>
          ))}
        </Box>
        <Box align="right" mr="8" mt="4">
          <Text fontWeight="500" fontSize="18px" color="#3870FF">
            {"Subtotal: " + formatMoney(order_subtotal_excl_tax, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="18px" color="#F54A49">
            {"Descuento: " + formatMoney(order_discount, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="28px" color="#3870FF">
            {"Total: " + formatMoney(order_total, CURRENCY_NAME)}
          </Text>
        </Box>
      </Box>
    </div>
  );
};
