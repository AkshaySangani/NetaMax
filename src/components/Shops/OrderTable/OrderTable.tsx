import React, { ReactElement } from "react";

import { CURRENCY_NAME } from "constants/localeFormatConstants";
import { IItem } from "dataflows/Shops/IItem";
import { screenSizes } from "styled/screen";
import { formatMoney } from "utils/currencyUtils";

import { Box, Container, Divider, Flex, Grid, GridItem, Text } from "@chakra-ui/layout";
import { useMediaQuery, Table, Tbody, Td, Tr } from "@chakra-ui/react";

import { IOrderTableProps } from "./IOrderTableProps";

/**
 * The order table component.
 * @param {IOrderTableProps} props the order table props.
 * @returns { ReactElement } The table component
 *
 */
export const OrderTable = (props: IOrderTableProps): ReactElement => {
  const { subtotal, total, discount, objects } = props;
  const [isLargerThanSm] = useMediaQuery(`(min-width: ${screenSizes.sm}px)`);

  /**
   * Renders the Order Table mobile component.
   * @returns {React.ReactElement} The NavBar mobile component
   */
  const renderMobileOrderTable = (): React.ReactElement => {
    return (
      <Box>
        {objects.map((object: IItem) => (
          <Box key={`${object.image}-${object.name}-${object.price}`} mb="4">
            <Divider mb="4" borderColor="#9F9F9F" />
            <Flex ml="4">
              <Box display="flex">
                <Text fontWeight="700" fontSize="19px">
                  {object.name}
                </Text>
              </Box>
            </Flex>
            <Grid ml="16" mt="2" h="40px" templateColumns="repeat(4, 1fr)">
              <GridItem colSpan={2}>
                <Text fontWeight="700" fontSize="17px">
                  Cantidad:
                </Text>
              </GridItem>
              <GridItem colSpan={2}>
                <Text fontWeight="700" fontSize="17px">
                  Precio:
                </Text>
              </GridItem>
              <GridItem colSpan={2}>
                <Text>{object.qty}</Text>
              </GridItem>
              <GridItem colSpan={2}>
                <Text>{formatMoney(object.price, CURRENCY_NAME)}</Text>
              </GridItem>
            </Grid>
          </Box>
        ))}
        <Box align="right" mr="8" mt="4">
          <Text fontWeight="700" fontSize="18px" color="#777">
            Subtotal: {formatMoney(subtotal, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="18px" color="#777">
            Descuento: {formatMoney(discount, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="30px" color="#3870FF">
            Total: {formatMoney(total, CURRENCY_NAME)}
          </Text>
        </Box>
      </Box>
    );
  };

  /**
   * Renders the Order Table mobile component.
   * @returns {React.ReactElement} The NavBar mobile component
   */
  const renderDesktopOrderTable = (): React.ReactElement => {
    return (
      <Container
        centerContent={false}
        maxW="976px"
        boxShadow="md"
        rounded="md"
        bg="white"
        display={["none", "block", "block", "block"]}
      >
        <Box maxHeight="320px" overflow="auto">
          <Table variant="simple">
            <Tbody>
              {objects.map((object: IItem) => (
                <Tr key={`${object.image}-${object.name}-${object.price}`}>
                  <Td>
                    <Text fontWeight="700">{object.name}</Text>
                  </Td>
                  <Td>
                    <Text fontWeight="700">Cantidad </Text>
                    {object.qty}
                  </Td>
                  <Td>
                    <Text fontWeight="700">Precio </Text>
                    {formatMoney(object.price, CURRENCY_NAME)}
                  </Td>
                </Tr>
              ))}
            </Tbody>
          </Table>
        </Box>
        <Box align="right" mr="8" mt="4" pb="4">
          <Text fontWeight="700" fontSize="18px" color="#777">
            Subtotal: {formatMoney(subtotal, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="18px" color="#777">
            Descuento: {formatMoney(discount, CURRENCY_NAME)}
          </Text>
          <Text fontWeight="700" fontSize="30px" color="#3870FF">
            Total: {formatMoney(total, CURRENCY_NAME)}
          </Text>
        </Box>
      </Container>
    );
  };

  return isLargerThanSm ? renderDesktopOrderTable() : renderMobileOrderTable();
};
