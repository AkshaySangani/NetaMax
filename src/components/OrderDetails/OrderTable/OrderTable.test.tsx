import { render, cleanup } from "@testing-library/react";
import { OrderTable } from "./OrderTable";
import { IOrderTableProps } from "./IOrderTable";

describe("OrderTable Component", () => {
  let mock: IOrderTableProps;

  beforeEach(() => {
    mock = {
      items: [
        {
          name: "Cigarros Pall Mall Mykonos",
          product_id: 460,
          quantity: 1,
          unit_price_incl_tax: 53.95,
        },
        {
          name: "Pasta Dental Colgate Triple Accion Original 50 M",
          product_id: 79,
          quantity: 2,
          unit_price_incl_tax: 13.5,
        },
      ],
      order_discount: 10,
      order_subtotal_excl_tax: 745.25,
      order_total: 735.25,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("OrderTable should be render correctly", () => {
    const { container } = render(<OrderTable {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Image should be render correctly if it's present in items array", () => {
    mock = {
      items: [
        {
          name: "Cigarros Pall Mall Mykonos",
          product_id: 460,
          quantity: 1,
          unit_price_incl_tax: 53.95,
          image: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002615_0.jpeg",
        },
        {
          name: "Pasta Dental Colgate Triple Accion Original 50 M",
          product_id: 79,
          quantity: 2,
          unit_price_incl_tax: 13.5,
          image: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002629_0.jpeg",
        },
      ],
      order_discount: 10,
      order_subtotal_excl_tax: 745.25,
      order_total: 735.25,
    };
    const { container } = render(<OrderTable {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
