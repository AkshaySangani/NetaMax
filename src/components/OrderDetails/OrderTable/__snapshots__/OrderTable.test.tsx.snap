// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTable Component Image should be render correctly if it's present in items array 1`] = `
<div>
  <div>
    <div
      class="chakra-container css-1l1bxqv"
    >
      <div
        class="css-2rotl5"
      >
        <table
          class="chakra-table css-0"
          role="table"
        >
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-0"
                role="gridcell"
              >
                <div
                  class="css-k008qs"
                >
                  <img
                    class="chakra-image css-1hi5ck8"
                    src="https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002615_0.jpeg"
                  />
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    Cigarros Pall Mall Mykonos
                  </p>
                </div>
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Cantidad 
                </p>
                1
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Precio unitario 
                </p>
                $53.95
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-0"
                role="gridcell"
              >
                <div
                  class="css-k008qs"
                >
                  <img
                    class="chakra-image css-1hi5ck8"
                    src="https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002629_0.jpeg"
                  />
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    Pasta Dental Colgate Triple Accion Original 50 M
                  </p>
                </div>
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Cantidad 
                </p>
                2
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Precio unitario 
                </p>
                $13.50
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        align="right"
        class="css-17dmoar"
      >
        <p
          class="chakra-text css-w0heo8"
        >
          Subtotal: $745.25
        </p>
        <p
          class="chakra-text css-6ylitu"
        >
          Descuento: $10.00
        </p>
        <p
          class="chakra-text css-1r08sim"
        >
          Total: $735.25
        </p>
      </div>
    </div>
    <div
      class="css-r5smlo"
    >
      <div
        class="css-1yv77fo"
      >
        <div
          class="css-1itv5e3"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-8gqyvx"
          />
          <div
            class="css-1u3f7f6"
          >
            <div
              class="css-k008qs"
            >
              <img
                class="chakra-image css-1hi5ck8"
                src="https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002615_0.jpeg"
              />
              <p
                class="chakra-text css-1mzljxq"
              >
                Cigarros Pall Mall Mykonos
              </p>
            </div>
          </div>
          <div
            class="css-l3vf2k"
          >
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Cantidad
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Precio
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                1
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                $53.95
              </p>
            </div>
          </div>
        </div>
        <div
          class="css-1itv5e3"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-8gqyvx"
          />
          <div
            class="css-1u3f7f6"
          >
            <div
              class="css-k008qs"
            >
              <img
                class="chakra-image css-1hi5ck8"
                src="https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002629_0.jpeg"
              />
              <p
                class="chakra-text css-1mzljxq"
              >
                Pasta Dental Colgate Triple Accion Original 50 M
              </p>
            </div>
          </div>
          <div
            class="css-l3vf2k"
          >
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Cantidad
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Precio
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                2
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                $13.50
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        align="right"
        class="css-dm83zx"
      >
        <p
          class="chakra-text css-1md0051"
        >
          Subtotal: $745.25
        </p>
        <p
          class="chakra-text css-1xn3crm"
        >
          Descuento: $10.00
        </p>
        <p
          class="chakra-text css-hmfnqz"
        >
          Total: $735.25
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`OrderTable Component OrderTable should be render correctly 1`] = `
<div>
  <div>
    <div
      class="chakra-container css-1l1bxqv"
    >
      <div
        class="css-2rotl5"
      >
        <table
          class="chakra-table css-0"
          role="table"
        >
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-0"
                role="gridcell"
              >
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    Cigarros Pall Mall Mykonos
                  </p>
                </div>
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Cantidad 
                </p>
                1
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Precio unitario 
                </p>
                $53.95
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-0"
                role="gridcell"
              >
                <div
                  class="css-k008qs"
                >
                  <p
                    class="chakra-text css-1vg6q84"
                  >
                    Pasta Dental Colgate Triple Accion Original 50 M
                  </p>
                </div>
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Cantidad 
                </p>
                2
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <p
                  class="chakra-text css-1vg6q84"
                >
                  Precio unitario 
                </p>
                $13.50
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        align="right"
        class="css-17dmoar"
      >
        <p
          class="chakra-text css-w0heo8"
        >
          Subtotal: $745.25
        </p>
        <p
          class="chakra-text css-6ylitu"
        >
          Descuento: $10.00
        </p>
        <p
          class="chakra-text css-1r08sim"
        >
          Total: $735.25
        </p>
      </div>
    </div>
    <div
      class="css-r5smlo"
    >
      <div
        class="css-1yv77fo"
      >
        <div
          class="css-1itv5e3"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-8gqyvx"
          />
          <div
            class="css-1u3f7f6"
          >
            <div
              class="css-k008qs"
            >
              <p
                class="chakra-text css-1mzljxq"
              >
                Cigarros Pall Mall Mykonos
              </p>
            </div>
          </div>
          <div
            class="css-l3vf2k"
          >
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Cantidad
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Precio
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                1
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                $53.95
              </p>
            </div>
          </div>
        </div>
        <div
          class="css-1itv5e3"
        >
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-8gqyvx"
          />
          <div
            class="css-1u3f7f6"
          >
            <div
              class="css-k008qs"
            >
              <p
                class="chakra-text css-1mzljxq"
              >
                Pasta Dental Colgate Triple Accion Original 50 M
              </p>
            </div>
          </div>
          <div
            class="css-l3vf2k"
          >
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Cantidad
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-eo8gwn"
              >
                Precio
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                2
              </p>
            </div>
            <div
              class="css-tuh9u2"
            >
              <p
                class="chakra-text css-l8qto0"
              >
                $13.50
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        align="right"
        class="css-dm83zx"
      >
        <p
          class="chakra-text css-1md0051"
        >
          Subtotal: $745.25
        </p>
        <p
          class="chakra-text css-1xn3crm"
        >
          Descuento: $10.00
        </p>
        <p
          class="chakra-text css-hmfnqz"
        >
          Total: $735.25
        </p>
      </div>
    </div>
  </div>
</div>
`;
